pallets/settlement/src/lib.rs

```diff
@@ -1421,17 +1644,15 @@ impl<T: Config> Module<T> {
         )?;
 
         let (total_leg_count, filtered_legs) =
-            Self::filtered_legs(id, &portfolios, max_legs_count)?;
+            Self::filtered_legs(&id, &portfolios, max_legs_count)?;
         with_transaction(|| {
             for (leg_id, leg_details) in filtered_legs {
-                if let Err(_) = Self::lock_via_leg(&leg_details) {
-                    // rustc fails to infer return type of `with_transaction` if you use ?/map_err here
-                    return Err(DispatchError::from(Error::<T>::FailedToLockTokens));
-                }
+                Self::lock_via_leg(&leg_details)?;
                 <InstructionLegStatus<T>>::insert(id, leg_id, LegStatus::ExecutionPending);
             }
             Ok(())
-        })?;
+        })
+        .map_err(|e: DispatchError| e)?;
```

Comment:
Now that we can have other errors when trying to lock tokens, this makes it a bit easier to know what's failing.